### PR TITLE
readline: Render completions using the entire line

### LINKED ISSUE #2865: Tab completion bug
I seem to be having a tab completion problem. It always happens with filenames with spaces in them. If I type the beginning of a filename (that has spaces) and tab, and that tab completion matches multiple files, it will add a single quote to the beginning, complete up to the filename collision, then when I type another letter to differentiate, it will fill the whole name in, with another single quote, and after what was already in place. 

This is explained much better with a screenshot.

image

OS:  macOS High Sierra 10.13.6 (17G65)

Just updated to newest master of xonsh, still have this issue.

### PR DESCRIPTION
**Pull request by daniel-shimon**
Since the completions we return are completed wrt. the completion prefix,
in order to handle spaced strings we need to use the entire line as the completion prefix.
For example:
    ls 'spaced
Can now correctly result in:
    ls 'spaced file' / ls 'spaced dir/'

Closes #2865

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: xonsh/readline_shell.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ def completedefault(self, prefix, line, begidx, endidx):
             multiline_text=prev_text + line,
             cursor_index=len(prev_text) + endidx,
         )
-        rtn_completions = _render_completions(completions, prefix, plen)
+        rtn_completions = _render_completions(completions, line, plen)
```
Comment (jnoortheen) [inline xonsh/readline_shell.py:456]: Is it possible to have tests for the rl-completions ? mocking could work or you can use `rl_shell` mock in `test_readline_shell.py` for testing.

**Comment by daniel-shimon** [comment]
I'll look into it 👍

**Review by gforsyth — APPROVED** [review]
The change looks good to me, I'll hold off on merging in case @daniel-shimon is planning to add some mocks for this

**Comment by gforsyth** [comment]
Hey @daniel-shimon -- gentle ping here.  Is there anything else you wanted to add to this PR?

**Comment by daniel-shimon** [comment]
> Hey @daniel-shimon -- gentle ping here.  Is there anything else you wanted to add to this PR?

Hi, sorry recently I don't have a lot of spare time to do xonsh stuff..

This PR has some problems which I addressed locally and added some half-baked tests.
I need to work out some details when I'll get to it.. For now this should be treated as a draft

**Comment by gforsyth** [comment]
Closing out old PRs, thanks!

**Pull request closed without merging**